tests/test_mysql.py

```diff
@@ -200,6 +200,84 @@ def test_insert_with_column_names(mysql_table, mysql_con):
     assert df.equals(df2)
 
 
+def test_upsert_distinct(mysql_table, mysql_con):
+    create_table_sql = (
+        f"CREATE TABLE test.{mysql_table} " "(c0 varchar(100) NULL, " "c1 INT DEFAULT 42 NULL, " "c2 INT NOT NULL);"
+    )
+    with mysql_con.cursor() as cursor:
+        cursor.execute(create_table_sql)
+        mysql_con.commit()
+
+    df = pd.DataFrame({"c0": ["foo", "bar"], "c2": [1, 2]})
+
+    wr.mysql.to_sql(
+        df=df, con=mysql_con, schema="test", table=mysql_table, mode="upsert_distinct", use_column_names=True
+    )
+    wr.mysql.to_sql(
+        df=df, con=mysql_con, schema="test", table=mysql_table, mode="upsert_distinct", use_column_names=True
+    )
+    wr.mysql.to_sql(
+        df=df, con=mysql_con, schema="test", table=mysql_table, mode="upsert_distinct", use_column_names=True
+    )
```

Comment:
Can we test a few more edge cases for all methods please? For instance what if the primary key values change.
```
    df = pd.DataFrame({"c0": ["foo", "bar"], "c2": [1, 2]})

    wr.mysql.to_sql(
        df=df, con=mysql_con, schema="test", table=mysql_table, mode="upsert_replace_into", use_column_names=True
    )
    df2 = wr.mysql.read_sql_table(con=mysql_con, schema="test", table=mysql_table)
    assert bool(len(df2) == 2)
    wr.mysql.to_sql(
        df=df, con=mysql_con, schema="test", table=mysql_table, mode="upsert_replace_into", use_column_names=True
    )
    df3 = pd.DataFrame({"c0": ["baz", "bar"], "c2": [3, 2]})
    wr.mysql.to_sql(
        df=df3, con=mysql_con, schema="test", table=mysql_table, mode="upsert_replace_into", use_column_names=True
    )
    df4 = wr.mysql.read_sql_table(con=mysql_con, schema="test", table=mysql_table)
    assert bool(len(df4) == 3)
```

Reply:
Yes, I will work through more scenarios. Should they be placed into the same test functions with additional assertions or would it be better to have separate functions for each case?

Reply:
I think additional assertions within the same test functions is fine unless you feel it's getting too big or the cases do not belong to the same logical test. Thanks!